extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -477,31 +472,81 @@ export class AmpAnalytics extends AMP.BaseElement {
     // Precedence is event.vars > trigger.vars > config.vars.
     // Nested expansion not supported.
     return expandTemplate(template, key => {
-      const match = key.match(/([^(]*)(\([^)]*\))?/);
-      const name = match[1];
-      const argList = match[2] || '';
+      const {name, argList} = this.getNameArgs_(key);
       let raw = (event && event['vars'] && event['vars'][name]) ||
           (trigger['vars'] && trigger['vars'][name]) ||
           (this.config_['vars'] && this.config_['vars'][name]) ||
           '';
+
+      // Values can also be arrays and objects. Don't expand them.
       if (typeof raw == 'string') {
         raw = this.expandTemplate_(raw, trigger, event, opt_iterations - 1);
       }
       const val = opt_encode ? this.encodeVars_(raw, name) : raw;
-      return val + argList;
+      return val ? val + argList : val;
     });
   }
 
   /**
+   * Returns an array containing two values: name and args parsed from the key.
+   *
+   * @param {string} key The key to be parsed.
+   * @return {Object<string>}
+   * @private
+   */
+  getNameArgs_(key) {
+    if (!key) {
+      return ['', ''];
+    }
+    const match = key.match(/([^(]*)(\([^)]*\))?/);
```

Comment:
It shouldn't be. If it is, that is an error that we would like to surface. Added a user.error.